src/doc-utils.js

```diff
@@ -120,6 +125,7 @@ function breakParentGroup(groupStack) {
 }
 
 function propagateBreaks(doc) {
+  var alreadyVisited = new WeakMap();
```

Comment:
Yeah, at first I thought the problem was going to be widespread around the entire codebase so I needed to make it global and not hold any strong reference. But it turned out to only be scoped to this function so a Map is enough indeed.